### DISCUSSION TITLE
Where's the bottleneck with sound_capacity & SoundLimitReached?

### DISCUSSION THREAD

**Discussion by justinfrench** [tesselode/kira#105] [Q&A]
Hi gang! First, this is an incredible library. I know it's designed for games, but I'm been using it for a real-time midi sampler instrument and it's fantastic. Tonight I decided to stress-test it a bit though, and hit a wall with `SoundLimitReached` that I would like to understand better.

From what I've learned, there is a `sound_capacity` in `Capacities` that can be configured, and defaults to `128`. When I configure a lower number, I get a predictable `SoundLimitReached` error that makes sense to me. However, at `128` or higher, there's a noticeable slowdown/distortion effect before the `SoundLimitReached` error.

This suggests to me that I'm hitting some other kind of bottleneck in Kira's code before I hit the `sound_capacity` limit of `128`. Any ideas where in the code I should be looking, or other settings I should be messing with (CpalBackendSettings maybe?)? I've gotten as far as I can with a source code read.

**Comment by tesselode** [comment]
You might just be maxing out your CPU. Can you see what your OS's CPU monitor says?
